roboteam_ai/src/conditions/Condition.cpp

```diff
@@ -6,5 +6,35 @@ namespace ai {
 Condition::Condition(std::string name, bt::Blackboard::Ptr blackboard)
         :bt::Leaf(name, blackboard) { }
 
+std::string Condition::node_name() {
+    return name;
+}
+
+void Condition::initialize() {
+    if (! properties->getString("ROLE").empty()) {
+        robot = getRobotFromProperties(properties);
+        ball = World::getBall(); // update ball position
+        if (! robot) return;
+        if (! ball) return;
```

Comment:
This allows all conditions only to run when the ball is there. is that what we want?

Reply:
I've tested it irl with ssl-vision and a ball:

There is always a ball, once there was one ball in the world, so shouldn't be a problem. 
In ssl-vision, as well as grsim, our world state has a ball area of 99999, so in the tests I put that at 99999 as well. Reason for the area in the first place, is that I needed a way to check if the ball is defined yet. I check if the area of the ball is not 0, because that means the ball is not defined yet